source/elements/oneMKL/source/domains/spblas/operations/spsv.rst

```diff
@@ -0,0 +1,341 @@
+           oneapi::mkl::sparse::dense_vector_handle_t x_handle,
+           oneapi::mkl::sparse::dense_vector_handle_t y_handle,
+           oneapi::mkl::sparse::spsv_alg              alg,
+           oneapi::mkl::sparse::spsv_descr_t          spsv_descr,
+           sycl::buffer<std::uint8_t, 1>              workspace,
+       );
+
+       sycl::event spsv_optimize(
+           sycl::queue                                &queue,
+           oneapi::mkl::transpose                     opA,
+           void*                                      alpha,
+           oneapi::mkl::sparse::matrix_view           A_view,
+           oneapi::mkl::sparse::matrix_handle_t       A_handle,
+           oneapi::mkl::sparse::dense_vector_handle_t x_handle,
+           oneapi::mkl::sparse::dense_vector_handle_t y_handle,
+           oneapi::mkl::sparse::spsv_alg              alg,
+           oneapi::mkl::sparse::spsv_descr_t          spsv_descr,
+           void*                                      workspace,
+           const std::vector<sycl::event>             &dependencies = {},
+       );
+
+       sycl::event spsv(
+           sycl::queue                                &queue,
+           oneapi::mkl::transpose                     opA,
+           void*                                      alpha,
+           oneapi::mkl::sparse::matrix_view           A_view,
+           oneapi::mkl::sparse::matrix_handle_t       A_handle,
+           oneapi::mkl::sparse::dense_vector_handle_t x_handle,
+           oneapi::mkl::sparse::dense_vector_handle_t y_handle,
+           oneapi::mkl::sparse::spsv_alg              alg,
+           oneapi::mkl::sparse::spsv_descr_t          spsv_descr,
+           const std::vector<sycl::event>             &dependencies = {},
+       );
+
+   }
+
+.. container:: section
+
+   .. rubric:: Notes
+
+   - ``spsv_buffer_size`` and ``spsv_optimize`` must be called at least once before ``spsv``
+     with the same arguments. ``spsv`` can then be called multiple times.
+   - The data of the handles can be reset-ed before each call to ``spsv``.
+   - ``spsv_optimize`` and ``spsv`` are asynchronous.
+   - The algorithm defaults to ``spsv_alg::default_alg`` if a backend does not
+     support the provided algorithm.
+
+   .. rubric:: Input Parameters
+
+   queue
+      The SYCL command queue which will be used for SYCL kernels execution.
+
+   opA
+      Specifies operation ``op()`` on the input matrix. The possible options are
+      described in :ref:`onemkl_enum_transpose` enum class.
+
+   alpha
+      Host or USM pointer representing :math:`\alpha`. The USM allocation can be
+      on the host or device. Must be of the same type than the handles' data
+      type.
+
+   A_view
+      Specifies which part of the handle should be read as described by
+      :ref:`onemkl_sparse_matrix_view`. ``A_view.type_view`` must be
+      ``matrix_descr::triangular`` or ``matrix_descr::diagonal``.
```

Comment:
```suggestion
      Specifies which part of the handle should be read as described by
      :ref:`onemkl_sparse_matrix_view`. ``A_view.type_view`` must be
      (a) ``matrix_descr::triangular`` with ``A_view.uplo_view`` being
      ``uplo::lower``/``uplo::upper``, or (b) ``matrix_descr::diagonal``.
```

Reply:
I don't think it is needed to add `with ``A_view.uplo_view`` being ``uplo::lower``/``uplo::upper`. `uplo_view` is initialized to `lower` by default in https://github.com/oneapi-src/oneAPI-spec/pull/522/files#diff-fd386f5e0051f416d505f3796cbbe921a8bc7f688f3d39748ae41df6f0f0d769R67 and there are only 2 possible values.